src/lib/bcmp/bcmp_ping.cpp

```diff
@@ -45,28 +45,28 @@ err_t bcmp_send_ping_request(uint64_t node_id, const ip_addr_t *addr, const uint
 
   // clear the expected payload
   if (_expected_payload != NULL) {
-    vPortFree(_expected_payload);
+    bm_free(_expected_payload);
     _expected_payload = NULL;
     _expected_payload_len = 0;
   }
 
   // Lets only copy the paylaod if it isn't NULL just in case
   if (payload != NULL && payload_len > 0) {
     memcpy(&echo_req->payload[0], payload, payload_len);
-    _expected_payload = static_cast<uint8_t *>(pvPortMalloc(payload_len));
+    _expected_payload = static_cast<uint8_t *>(bm_malloc(payload_len));
     memcpy(_expected_payload, payload, payload_len);
     _expected_payload_len = payload_len;
   }
 
   printf("PING (%016" PRIx64 "): %" PRIu16 " data bytes\n", echo_req->target_node_id,
          echo_req->payload_len);
 
-  err_t rval =
+  BmErr rval =
       bcmp_tx(addr, BcmpEchoRequestMessage, reinterpret_cast<uint8_t *>(echo_req), echo_len, 0);
 
-  _ping_request_time = uptimeGetMicroSeconds();
+  _ping_request_time = bm_ticks_to_ms(bm_get_tick_count());
```

Comment:
Will this be a problem seeing as it is milliseconds?

Reply:
The printout is in milliseconds, and realistically this could be user defined in the future as right now the only way to get this is through CLI. So I don't think it's a problem to convert the time from micro to millis.